plugins/goodix-tp/fu-goodixtp-firmware.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2023 Goodix.inc <xulinkun@goodix.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include "fu-goodixtp-common.h"
+#include "fu-goodixtp-firmware.h"
+
+#define MAX_CHUNK_NUM 80
+
+struct FuGoodixChunkInfo {
+	guint8 type;
+	guint32 flash_addr;
+};
+
+typedef struct {
+	FuFirmwareClass parent_instance;
+	guint32 version;
+	guint8 *fw_data;
+	gint fw_len;
+	gint index;
+	struct FuGoodixChunkInfo chunk_info[MAX_CHUNK_NUM];
```

Comment:
this seems like a big thing to put on the stack; can you make this a GPtrArray please.